packages/application-shell/src/components/setup-flop-flip-provider/setup-flop-flip-provider.tsx

```diff
@@ -54,23 +91,44 @@ export const SetupFlopFlipProvider = (props: Props) => {
 
   const adapterArgs = React.useMemo(
     () => ({
-      sdk: {
-        // Allow to overwrite the client ID, passed via the `additionalEnv` properties
-        // of the application config.
-        // This is mostly useful for internal usage on our staging environments.
-        clientSideId: props.ldClientSideId ?? ldClientSideIdProduction,
-      },
-      flags,
       user: {
         key: props.user?.id,
-        custom: omitEmpty<CustomLDUser, Partial<CustomLDUser>>({
-          id: props.user?.launchdarklyTrackingId,
-          project: props.projectKey,
-          team: props.user?.launchdarklyTrackingTeam,
-          group: props.user?.launchdarklyTrackingGroup,
-          subgroup: props.user?.launchdarklyTrackingSubgroup,
-          tenant: props.user?.launchdarklyTrackingTenant,
-        }),
+      },
+      memory: {
+        user: {
+          key: props.user?.id,
+        },
+      },
+      launchdarkly: {
+        sdk: {
+          // Allow to overwrite the client ID, passed via the `additionalEnv` properties
+          // of the application config.
+          // This is mostly useful for internal usage on our staging environments.
+          clientSideId: props.ldClientSideId ?? ldClientSideIdProduction,
+        },
+        flags,
+        user: {
+          key: props.user?.id,
+          custom: omitEmpty<CustomLDUser, Partial<CustomLDUser>>({
+            id: props.user?.launchdarklyTrackingId,
+            project: props.projectKey,
+            team: props.user?.launchdarklyTrackingTeam,
+            group: props.user?.launchdarklyTrackingGroup,
+            subgroup: props.user?.launchdarklyTrackingSubgroup,
+            tenant: props.user?.launchdarklyTrackingTenant,
+          }),
+        },
+      },
+      graphql: {
+        user: {
+          key: props.user?.id,
+        },
+        uri: `${getMcApiUrl()}/graphql`,
+        query: 'query AllFeatures { allFeatures { name value } }',
```

Comment:
Can we put this into a `.graphql` file?

Reply:
What would be the advantage or it? It's a "raw" fetch in the end.

Reply:
So that we can validate the query

Reply:
Oh well.